--- a/dwl.c
+++ b/dwl.c
@@ -163,6 +163,10 @@ typedef struct {
 	unsigned int tags;
 	int isfloating;
 	int monitor;
+	int x;
+	int y;
+	int w;
+	int h;
 } Rule;
 
 /* Used to move all of the data necessary to render a surface from the top-level
@@ -357,6 +361,11 @@ applyrules(Client *c)
 			wl_list_for_each(m, &mons, link)
 				if (r->monitor == i++)
 					mon = m;
+			if (c->isfloating)
+				resize(c, r->x > 0 ? r->x + mon->w.x : mon->w.width / 2 - c->geom.width / 2 + mon->w.x,
+						r->y > 0 ? r->y + mon->w.y : mon->w.height / 2 - c->geom.height / 2 + mon->w.y,
+						r->w > 0 ? r->w : c->geom.width,
+						r->h > 0 ? r->h : c->geom.height, 1);
 		}
 	}
 	setmon(c, mon, newtags);
